### PR TITLE
ENT-8229 Skip invalid feeder dump files during import

### PR DESCRIPTION
**Pull request by craigcomstock**
Feeder dump files have in their name the hostkey of the feeder.
If this hostkey is not present in the __hubs table then skip
that dump file and log a warning recommending either to delete
the file or add the feeder to superhub.

Ticket: ENT-8229
Changelog: title

### CODE REVIEW

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please, thanks.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7406/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7406/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7406/

File: templates/federated_reporting/import.sh [hunk 1]
```diff
@@ -52,9 +52,14 @@ declare -a hostkeys
 for file in $dump_files; do
   hostkey=$(basename "$file" | cut -d. -f1)
   hostkeys+=($hostkey)
-  "$CFE_BIN_DIR"/psql -U $CFE_FR_DB_USER -d cfdb --set "ON_ERROR_STOP=1" \
-                      -c "SELECT ensure_feeder_schema('$hostkey', ARRAY[$table_whitelist]);" \
-    > schema_setup.log 2>&1 || failed=1
+  if [ -z $(psql --csv --tuples-only -U cfapache -d cfdb -c "SELECT hub_id FROM __hubs WHERE hostkey = '$hostkey';") ]; then
+    log "No feeder with hostkey $hostkey found in cfdb.__hubs, skipping the dump file $file, consider deleting this file or re-adding the feeder to superhub"
+    dump_files=$(echo "$dump_files" | sed "s,\s\?$file,," | xargs)
```
Comment (craigcomstock) [inline templates/federated_reporting/import.sh:57]: `alias trim=xargs`, why can't I just have a command called `trim` :p

Comment (vpodzime) [reply templates/federated_reporting/import.sh:57]: But is it really needed? `$dump_files` has the files on separate lines, right? Why should we change it here?

**Review by vpodzime — APPROVED** [review]
Looks good to me either way. Good stuff, thanks!

File: templates/federated_reporting/import.sh (hunk 1 above)
Comment (craigcomstock) [reply templates/federated_reporting/import.sh:57]: When I was testing I observed the files are space separated so was absolutely required if I were to try remove the first or last files.

**Comment by craigcomstock** [comment]
Not sure why sequential-tests failed, was known invalid argument problem. I suggest we merge. @vpodzime yes?

**Comment by craigcomstock** [comment]
cherry picks
3.18 https://github.com/cfengine/masterfiles/pull/2182
3.15 https://github.com/cfengine/masterfiles/pull/2183

**Pull request merged**